RecoTracker/TkSeedingLayers/test/test_catch2_SeedingHitSet.cpp

```diff
@@ -0,0 +1,78 @@
+#include "RecoTracker/TkSeedingLayers/interface/SeedingHitSet.h"
+#include "DataFormats/TrackerRecHit2D/interface/TrackerSingleRecHit.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHit.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiStripRecHit2D.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiStripRecHit1D.h"
+#include "DataFormats/TrackerRecHit2D/interface/MTDTrackingRecHit.h"
+#include "DataFormats/TrackerRecHit2D/interface/Phase2TrackerRecHit1D.h"
+
+#include "catch.hpp"
+
+#include <iostream>
+
+TEST_CASE("SeedingHitSet testing constructors", "[SeedingHitSet]") {
+  using Hit = SeedingHitSet::RecHit;
+  using HitPointer = SeedingHitSet::ConstRecHitPointer;
+
+  HitPointer hitOne = new SiPixelRecHit();
+  HitPointer hitTwo = new MTDTrackingRecHit();
+  HitPointer hitThree = new SiStripRecHit1D();
+  HitPointer hitFour = new SiStripRecHit2D();
+  HitPointer hitFive = new Phase2TrackerRecHit1D();
+
+  HitPointer HIT_NULL = nullptr;
+
+  SECTION("Check vector constructor") {
+    std::vector<HitPointer> containerOne = {hitOne, hitTwo, hitThree, hitFour, hitFive};
+    std::vector<HitPointer> containerTwo = {hitOne, hitTwo, hitThree, HIT_NULL, hitFive};
+    std::vector<HitPointer> containerThree = {hitOne, HIT_NULL, hitThree, hitFour, hitFive};
+
+    SeedingHitSet vecOne(containerOne);
+    REQUIRE(vecOne.size() == containerOne.size());
+    SeedingHitSet vecTwo(containerTwo);
+    REQUIRE(vecTwo.size() == 3);
+    SeedingHitSet vecThree(containerThree);
+    REQUIRE(vecThree.size() == 0);
+  }
+
+  SECTION("Check braced constructor") {
+    std::vector<HitPointer> containerOne = {hitOne, hitTwo, hitThree, hitFour, hitFive};
```

Comment:
Do we need `containerOne, containerTwo, containerThree` defined in this section?

Reply:
Actually no, removing them.